### DISCUSSION TITLE
Object, Process and Data Models for Musescore's code ?

### DISCUSSION THREAD

**Discussion by ja2777** [musescore/MuseScore#25906] [General]
Hi in search of Object, Process and Data Models for Musescore's code. Pretty sure someone would have done some at some point. My name search did not come up with any or I did not use the right search wildcards with the names. Thanks for your time.

**Comment by Thomas-Ganter** [comment]
I have the same issue. 

I was working on fixing some issues and 98% of my time is spent understanding what does what in the code. 
Comments are sparse, and context is rarely given. 

While the Codebase is clean and most of the names are highly self-explanatory, it still is somewhat of a nightmare identifying where to tackle a feature ... 8^(

@its-not-nice , you seem to be one of the CodeWarriors of MuseScore  --  any insights on this?

**Reply by ghost** [reply]
I'm no code warrior, I just make decisions about engraving stuff :)
But we do have many developers who, no doubt, have their own opinions about the state of the codebase and should be able to chime in.

**Reply by Thomas-Ganter** [reply]
Okay, who is the head Code Conductor then to chime in here?

**Comment by ja2777** [comment]
No man, I gave up and just in dealing with it like it is, I'd love to
modify the code and make it not a bunch of spaghetti stuff thrown together
and make it commented with Object oriented design methodologies. But you
know, when something's such a mess that you can't even install it and step
through it with a debugger. You have to waste too much time in
it.......Life is too short.

**Reply by Thomas-Ganter** [reply]
I could install it and I could step through the code. 
The setup instructions were easy to follow and correct.

**Comment by ja2777** [comment]
Then it must have changed from when I looked at it. And I'm sure they did a
really good job of object-oriented programming and commenting their code
now that wasn't there before when I wasted my time with it.
